hls4ml/templates/vivado/nnet_utils/nnet_activation_stream.h

```diff
@@ -74,6 +74,53 @@ void relu(hls::stream<data_T> &data, hls::stream<res_T> &res) {
     }
 }
 
+// *************************************************
+//       Sigmoid Activation
+// *************************************************
+
+template<class data_T, class res_T, typename CONFIG_T>
+void sigmoid(hls::stream<data_T> &data, hls::stream<res_T> &res) {
+    // Initialize the lookup table
+#ifdef __HLS_SYN__
+    bool initialized = false;
+    typename CONFIG_T::table_t sigmoid_table[CONFIG_T::table_size];
+#else
+    static bool initialized = false;
+    static typename CONFIG_T::table_t sigmoid_table[CONFIG_T::table_size];
+#endif
+    if (!initialized) {
+        init_sigmoid_table<CONFIG_T, CONFIG_T::table_size>(sigmoid_table);
+        initialized = true;
+    }
+
+    int data_round;
```

Comment:
Can we get rid of these and declare them in the loop? I was told this style was needed for some ancient version of Vivado but I have never encountered this issue.